nm: Do not use `Other` interface type

cathay4t opened the pull request:
For OVS geneve interface, nmstate is incorrect showing it as managed
even NM marked it as unmanaged.

The root cause is NM is using `InterfaceType::Other("generic")` for this
kind of interface, while nispor is showing it as Ethernet interface.
Since `InterfaceType::Other` is treated as user space interface, the
merging will not working correctly.

To fix the issue, for NM unknown interface, we set the interface type to
`InterfaceType::Unknown` which is a kernel interface.

Reference bug: https://bugzilla.redhat.com/show_bug.cgi?id=2232578

Tier1 test case included.

liangwen12year commented:
The test file `tests/integration/ovs_test.py` needs to be reformated by pylint.

cathay4t committed:
nm: Do not use `Other` interface type

For OVS geneve interface, nmstate is incorrect showing it as managed
even NM marked it as unmanaged.

The root cause is NM is using `InterfaceType::Other("generic")` for this
kind of interface, while nispor is showing it as Ethernet interface.
Since `InterfaceType::Other` is treated as user space interface, the
merging will not working correctly.

To fix the issue, for NM unknown interface, we set the interface type to
`InterfaceType::Unknown` when it has MAC address.

Reference bug: https://bugzilla.redhat.com/show_bug.cgi?id=2232578

Tier1 test case included.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
> 

Done

cathay4t commented:
Sorry, I have waited long enough. And this is import use case for OpenShift with OVN.
